shell/app/modules/application/common/app-selector.tsx

```diff
@@ -14,24 +14,53 @@
 import * as React from 'react';
 import { LoadMoreSelector, Icon as CustomIcon } from 'common';
 import { goTo } from 'common/utils';
-import { map } from 'lodash';
+import { map, isArray, filter, isEmpty, find, get } from 'lodash';
 import { Tooltip } from 'app/nusi';
+import i18n from 'i18n';
 import { getApps } from 'common/services';
 import routeInfoStore from 'app/common/stores/route';
 import appStore from 'application/stores/application';
+import { getAppDetail } from 'application/services/application';
 import './app-selector.scss';
 
 interface IProps{
   [pro: string]: any;
   value: string | number;
   projectId?: string;
-  onClickItem: (arg?: any) => void;
+  onClickItem?: (arg: IApplication) => void;
+  onChange?: (arg: number) => void;
 }
 
 const AppItem = (app: IApplication) => {
   return <Tooltip key={app.id} title={app.name}>{app.displayName || app.name}</Tooltip>;
 };
 
+
+export const chosenItemConvert = (values: any) => {
+  const curApp = appStore.getState(s => s.detail);
+  const curValues = isArray(values) ? values : (values && [values]);
+  const existApp = {};
+  if(curApp.id) existApp[curApp.id] = curApp;
+  const reValue = map(curValues, item => {
+    const appItem = existApp[item.value] || {};
+    return item.label ? item : { ...appItem, ...item, label: appItem.displayName || appItem.name };
```

Comment:
if `appItem` is `{}`. label would be undefined, is this in line with expectations?